apps/src/templates/instructions/TopInstructionsHeader.jsx

```diff
@@ -0,0 +1,246 @@
+    wrapper: {
+      float: 'right'
+    },
+    button: {
+      height: 24,
+      marginTop: '3px',
+      marginBottom: '3px'
+    },
+    buttonImg: {
+      lineHeight: '24px',
+      fontSize: 15,
+      paddingLeft: 12
+    }
+  },
+  helpTabsRtl: {
+    float: 'right',
+    paddingTop: 6,
+    paddingRight: 30
+  },
+  helpTabs: {
+    float: 'left',
+    paddingTop: 6,
+    paddingLeft: 30
+  },
+  collapserIcon: {
+    showHideButton: {
+      position: 'absolute',
+      top: 0,
+      margin: 0,
+      lineHeight: styleConstants['workspace-headers-height'] + 'px',
+      fontSize: 18,
+      ':hover': {
+        cursor: 'pointer',
+        color: color.white
+      }
+    },
+    showHideButtonLtr: {
+      left: 8
+    },
+    showHideButtonRtl: {
+      right: 8
+    },
+    teacherOnlyColor: {
+      color: color.lightest_cyan,
+      ':hover': {
+        cursor: 'pointer',
+        color: color.default_text
+      }
+    }
+  }
+};
+
+function TopInstructionsHeader(props) {
+  const {
+    teacherOnly,
+    tabSelected,
+    isCSDorCSP,
+    displayHelpTab,
+    displayFeedback,
+    displayKeyConcept,
+    isViewingAsTeacher,
+    fetchingData,
+    handleDocumentationClick,
+    handleInstructionTabClick,
+    handleHelpTabClick,
+    handleCommentTabClick,
+    handleTeacherOnlyTabClick,
+    handleClickCollapser,
+
+    // redux provided
+    isMinecraft,
+    ttsLongInstructionsUrl,
+    hasContainedLevels,
+    isRtl,
+    documentationUrl,
+    teacherMarkdown,
+    isEmbedView,
+    isCollapsed
+  } = props;
+
+  const showContainedLevelAnswer =
+    hasContainedLevels && $('#containedLevelAnswer0').length > 0;
+
+  const collapserIconStyles = {
+    ...styles.collapserIcon.showHideButton,
+    ...(isRtl
+      ? styles.collapserIcon.showHideButtonRtl
+      : styles.collapserIcon.showHideButtonLtr),
+    ...(teacherOnly && styles.collapserIcon.teacherOnlyColor)
+  };
+
+  return (
+    <PaneHeader
+      hasFocus={false}
+      teacherOnly={teacherOnly}
+      isMinecraft={isMinecraft}
+    >
+      <div style={styles.paneHeaderOverride}>
+        {/* For CSF contained levels we use the same audio button location as CSD/CSP*/}
+        {tabSelected === TabType.INSTRUCTIONS &&
+          ttsLongInstructionsUrl &&
+          (hasContainedLevels || isCSDorCSP) && (
+            <InlineAudio
+              src={ttsLongInstructionsUrl}
+              style={isRtl ? styles.audioStyleRTL : styles.audioStyle}
+              autoplayTriggerElementId="codeApp"
+            />
+          )}
+        {documentationUrl && tabSelected !== TabType.COMMENTS && (
+          <PaneButton
+            iconClass="fa fa-book"
+            label={i18n.documentation()}
+            isRtl={isRtl}
+            headerHasFocus={false}
+            onClick={handleDocumentationClick}
+            isMinecraft={isMinecraft}
+          />
+        )}
+        <div style={isRtl ? styles.helpTabsRtl : styles.helpTabs}>
+          <InstructionsTab
```

Comment:
if you create a `renderInstructionsTab` helper i think you could seriously cut down on the number of lines in this component.